### PR TITLE
Refactor: move generic functions of integration test to util directory

### LINKED ISSUE #87427: [TaintBasedEvictions] Ensure test coverage is complete and up-to-date
As part of promoting TaintBasedEvictions to GA in 1.18 (https://github.com/kubernetes/kubernetes/issues/87161), we need to make sure that the test coverage is sufficient. There are some tests mentioned in the KEP, and as mentioned in https://github.com/kubernetes/enhancements/pull/1450#discussion_r366899543, those that are still under scheduling should be moved to the appropriate SIGs (sig-node or sig-cloud-provider)

/kind cleanup

### PR DESCRIPTION
**Pull request by fengzixu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

According to the suggestion from @smarterclayton on this PR https://github.com/kubernetes/kubernetes/pull/88152, we need to split cleanup task into two PRs. one is moving generic functions(which I submited in this PR), another one is moving related tests from sig-scheduling to sig-node.

**Which issue(s) this PR fixes**:

Fixes #87427

**Special notes for your reviewer**:
```release-note
NONE
```

**Does this PR introduce a user-facing change?**:

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by fengzixu** [comment]
@mattjmcnaughton  @smarterclayton @damemi Ping

**Comment by fengzixu** [comment]
@smarterclayton @mattjmcnaughton  PTAL

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

on scheduling side.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

**Comment by fengzixu** [comment]
@sig-testing-approvers

**Comment by fengzixu** [comment]
@liggitt @SataQiu PTAL, if you are available.

File: test/integration/util/util.go [hunk 1]
```diff
@@ -131,3 +146,326 @@ func StartFakePVController(clientSet clientset.Interface) ShutdownFunc {
 	informerFactory.Start(ctx.Done())
 	return ShutdownFunc(cancel)
 }
+
+// TestContext store necessary context info
+type TestContext struct {
```
Comment (SataQiu) [inline test/integration/util/util.go]: SGTM! But different components have different `TestContext`, such as:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/garbagecollector/garbage_collector_test.go#L200-L210
```go
type testContext struct {
	tearDown           func()
	gc                 *garbagecollector.GarbageCollector
	clientSet          clientset.Interface
	apiExtensionClient apiextensionsclientset.Interface
	dynamicClient      dynamic.Interface
	metadataClient     metadata.Interface
	startGC            func(workers int)
	// syncPeriod is how often the GC started with startGC will be resynced.
	syncPeriod time.Duration
}
```
https://github.com/kubernetes/kubernetes/blob/master/test/integration/apiserver/admissionwebhook/admission_test.go#L69-L80

```go
type testContext struct {
	t *testing.T

	admissionHolder *holder

	client    dynamic.Interface
	clientset clientset.Interface
	verb      string
	gvr       schema.GroupVersionResource
	resource  metav1.APIResource
	resources map[schema.GroupVersionResource]metav1.APIResource
}
```
https://github.com/kubernetes/kubernetes/blob/master/test/integration/volumescheduling/util.go#L42-L52

```go
type testContext struct {
	closeFn         framework.CloseFunc
	httpServer      *httptest.Server
	ns              *v1.Namespace
	clientSet       *clientset.Clientset
	informerFactory informers.SharedInformerFactory
	scheduler       *scheduler.Scheduler

	ctx      context.Context
	cancelFn context.CancelFunc
}
```

These `testContext` are very different. I'm a little concerned about whether it's appropriate to make `TestContext` as a public dependency.

Comment (fengzixu) [reply test/integration/util/util.go]: Actually, I am moving the util functions of sig-scheduling and sig-node integration test. For different definitions of `testContext`, we can permit different sig has their own definition. If they decide to move it into the util. they can adjust the definition of `TestContext` on util to adapt.

This PR is just for adapting changes of moving TainBasedEviction test from sig-scheduling to sig-node. https://github.com/kubernetes/kubernetes/issues/87427

**Pull request merged**